plugins/device/cmdexec/device.go

```diff
@@ -0,0 +1,51 @@
+package cmdexec
+
+import (
+	"os/exec"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func createVgName(device string) string {
+	vgName := strings.Replace("vg"+device, "/", "-", -1)
+	return vgName
+}
+
+// DeviceSetup is used to prepare device before using devices.
+func DeviceSetup(device string) error {
+
+	var err error
+	pvcreateCmd := exec.Command("pvcreate", "--metadatasize=128M", "--dataalignment=256K", device)
+	if err := pvcreateCmd.Run(); err != nil {
+		log.WithError(err).WithField("device", device).Error("pvcreate failed for device")
+		return err
+	}
+	vgName := createVgName(device)
+	vgcreateCmd := exec.Command("vgcreate", vgName, device)
+	if err = vgcreateCmd.Run(); err != nil {
+		log.WithError(err).WithField("device", device).Error("vgcreate failed for device")
+		return err
+	}
+
+	defer func() {
```

Comment:
This defer is useless as the call it's at the end of the function. If there's an error anywhere in code above, control won't even reach here.